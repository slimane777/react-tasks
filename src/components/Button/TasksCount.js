import React from 'react'
import PropTypes from 'prop-types'
import { Count } from './styles'

const TasksCount = ({ count }) => {
  return (
    <Count>
        {count} Tasks
    </Count>
  )
}

TasksCount.propTypes = {
    count: PropTypes.number.isRequired
}

export default TasksCount;