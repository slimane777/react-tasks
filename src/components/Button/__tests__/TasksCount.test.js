import React from 'react';
import renderer from 'react-test-renderer';
import TasksCount from '../TasksCount';

import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('TasksCount test', () => {
  const initialState = {tasks:{tasks: []}}
  const mockStore = configureStore()
  let store,wrapper

  it('TasksCount should match snapshot', () => {
    store = mockStore(initialState)
    const component = renderer.create(<Provider store={store}><TasksCount
      count={() => {}} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
