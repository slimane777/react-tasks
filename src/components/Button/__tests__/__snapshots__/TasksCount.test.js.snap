// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksCount test TasksCount should match snapshot 1`] = `
<span
  className="sc-gtsrHT jOoloQ"
>
   Tasks
</span>
`;
